app/src/androidTest/java/ch/epfl/polychef/IngredientTests.java

```diff
@@ -0,0 +1,65 @@
+package ch.epfl.polychef;
+
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+
+import ch.epfl.polychef.recipe.Ingredient;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class IngredientTests {
+
+    @Test
+    public void ingredientsRejectsInvalidArguments() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> new Ingredient("", 50, Ingredient.Unit.TEASPOON));
+        Assertions.assertThrows(IllegalArgumentException.class, () -> new Ingredient("Carrots", -5, Ingredient.Unit.TEASPOON));
+    }
+
+    @Test
+    public void gettersReturnCorrectAttributes(){
+        Ingredient ingredient = new Ingredient("Carrots", 300, Ingredient.Unit.TABLESPOON);
+        assertEquals(ingredient.getName(), "carrots");
+        assertEquals(ingredient.getUnit(), Ingredient.Unit.TABLESPOON);
+        assertTrue(ingredient.getQuantity() == 300d);
```

Comment:
We should use assertThat or assertEquals for better error message instead of assertTrue (same multiple time in this file)